types/pouchdb-core/index.d.ts

```diff
@@ -555,9 +550,9 @@ declare namespace PouchDB {
          *
          * @see {@link https://pouchdb.com/2014/06/17/12-pro-tips-for-better-code-with-pouchdb.html|PouchDB Pro Tips}
          */
-        post(doc: Core.PostDocument<Content>,
-            options: Core.PostOptions | null,
-            callback: Core.Callback<Core.Error, Core.Response>): void;
+        post<Model>(doc: Core.PostDocument<Content & Model>,
+                    options: Core.Options | null,
+                    callback: Core.Callback<Core.Response>): void;
```

Comment:
Which is to say, now: `Core.Callback<ReturnType> = (error: Core.Error, result: ReturnType) => void`, so it is still the Node-style callback you expect it to be.